sdk/src/wallet/migration/chrysalis.rs

```diff
@@ -0,0 +1,315 @@
+pub(crate) struct AccountDetailsDto {
+    pub(crate) index: u32,
+    coin_type: u32,
+    alias: String,
+    public_addresses: Vec<AccountAddress>,
+    internal_addresses: Vec<AccountAddress>,
+    addresses_with_unspent_outputs: Vec<Value>,
+    outputs: HashMap<String, Value>,
+    locked_outputs: HashSet<String>,
+    unspent_outputs: HashMap<String, Value>,
+    transactions: HashMap<String, Value>,
+    pending_transactions: HashSet<String>,
+    incoming_transactions: HashMap<String, Value>,
+    native_token_foundries: HashMap<String, Value>,
+}
+
+pub async fn migrate_db_chrysalis_to_stardust(
+    storage_path: impl Into<String> + Send,
+    password: Option<Password>,
+) -> Result<()> {
+    let storage_path_string = storage_path.into();
+    let storage_path = Path::new(&storage_path_string);
+    // `/db` will be appended to the chrysalis storage path, because that's how it was done in the chrysalis wallet
+    let chrysalis_storage_path = &(*storage_path).join("db");
+
+    let chrysalis_data = get_chrysalis_data(chrysalis_storage_path, password)?;
+
+    // create new accounts base on previous data
+    let (new_accounts, secret_manager_dto) = migrate_from_chrysalis_data(&chrysalis_data, storage_path, false)?;
+
+    // convert to string keys
+    let chrysalis_data_with_string_keys = chrysalis_data
+        .into_iter()
+        .map(|(k, v)| {
+            Ok((
+                String::from_utf8(k).map_err(|_| Error::Migration("invalid utf8".into()))?,
+                v,
+            ))
+        })
+        .collect::<Result<HashMap<String, String>>>()?;
+
+    log::debug!(
+        "Chrysalis data: {}",
+        serde_json::to_string_pretty(&chrysalis_data_with_string_keys)?
+    );
+
+    let mut opts = Options::default();
+    opts.create_if_missing(true);
+    opts.create_missing_column_families(true);
+    let stardust_db = DB::open(&opts, storage_path).unwrap();
+
+    // store chrysalis data in a new key
+    stardust_db.put(
+        CHRYSALIS_STORAGE_KEY,
+        serde_json::to_string(&chrysalis_data_with_string_keys)?,
+    )?;
+    // write new accounts to db (with account indexation)
+    let accounts_indexation_data: Vec<u32> = new_accounts.iter().map(|account| account.index).collect();
+    stardust_db.put(
+        ACCOUNTS_INDEXATION_KEY,
+        serde_json::to_string(&accounts_indexation_data)?,
+    )?;
+    for new_account in new_accounts {
+        stardust_db.put(
+            &format!("{ACCOUNT_INDEXATION_KEY}{}", new_account.index),
+            serde_json::to_string(&new_account)?,
+        )?;
+    }
+
+    if let Some(secret_manager_dto) = secret_manager_dto {
+        // This is required for the secret manager to be loaded
+        stardust_db.put(WALLET_INDEXATION_KEY, format!("{{ \"coinType\": {IOTA_COIN_TYPE}}}"))?;
+        stardust_db.put(SECRET_MANAGER_KEY, secret_manager_dto)?;
+    }
+
+    // set db migration version
+    let migration_version = MigrationVersion {
```

Comment:
Reuse the existing one from migrate_4

Reply:
You mean use the consts from here
```
impl MigrationData for Migrate {
    const ID: usize = 4;
    const SDK_VERSION: &'static str = "1.0.0-rc.0";
    const DATE: time::Date = time::macros::date!(2023 - 07 - 19);
```
? Because there is no built MigrationVersion

Reply:
Done